Source/engine/assets.cpp

```diff
@@ -478,11 +485,24 @@ void UnloadModArchives()
 
 void LoadModArchives(std::span<const std::string_view> modnames)
 {
+	std::string targetPath;
+	for (std::string_view modname : modnames) {
+		targetPath = StrCat(paths::PrefPath(), "mods/", modname) + DirectorySeparator;
+		if (FileExists(targetPath)) {
+			OverridePaths.emplace_back(targetPath);
+		}
+		targetPath = StrCat(SDL_GetBasePath(), "mods/", modname) + DirectorySeparator;
+		if (FileExists(targetPath)) {
+			OverridePaths.emplace_back(targetPath);
+		}

```

Comment:
```suggestion
		targetPath = StrCat(paths::PrefPath(), "mods" DIRECTORY_SEPARATOR_STR, modname, DirectorySeparator);
		if (FileExists(targetPath)) {
			OverridePaths.emplace_back(targetPath);
		}
		targetPath = StrCat(SDL_GetBasePath(), "mods" DIRECTORY_SEPARATOR_STR, modname, DirectorySeparator);
		if (FileExists(targetPath)) {
			OverridePaths.emplace_back(targetPath);
		}
```

Reply:
This will cause a bug sine StrCat() sees DirectorySeparator as an int8 ("47") and not a character.

Reply:
Oh right cause it's a char. We can use `DIRECTORY_SEPARATOR_STR` instead.